### PR TITLE
upgrade GCP CloudRun TF module

### PR DESCRIPTION
**Pull request by yossi-lavi**

### CODE REVIEW

File: gcp-cloud-run/01-variables.tf [hunk 1]
```diff
@@ -95,64 +115,29 @@ variable "firewall" {
...
+variable "pvault_region" {
   description = "Vault Region. if empty fallback to default region"
   type        = string
+  default     = null
 }
 
-variable "image" {
-  description = "Vault server image name"
+variable "pvault_repository" {
+  description = "Vault server repository name"
```
Comment (nirhaas) [inline gcp-cloud-run/01-variables.tf:135]:
```suggestion
  description = "Vault Server container repository"
```

File: gcp-cloud-run/01-variables.tf [hunk 2]
```diff
@@ -167,119 +152,153 @@ variable "pvault_log_customer_env" {
...
 }
 
-########################
-### Vault CLI Server ###
-########################
-variable "vault_cli_subnet" {
-  description = "Subnet where Vault CLI will be deployed"
+###########################
+### Cloud Load Balancer ###
+###########################
```
Comment (nirhaas) [inline gcp-cloud-run/01-variables.tf:184]: Will those resources be created if the proxy is not created? If not, better clarify the description

File: gcp-cloud-run/10-network.tf [hunk 3]
```diff
@@ -112,9 +118,10 @@ resource "google_project_service" "vpcaccess_api" {
 resource "google_vpc_access_connector" "connector_vault_cloud_run" {
   name     = "${var.deployment_id}-sql-cn"
   provider = google-beta
-  region   = local.vault_region
+  region   = local.pvault_region
+  network = var.create_vpc ? null : var.vpc_id
   subnet {
-    name = module.vpc.subnets["${local.vault_region}/${var.deployment_id}-sb-connector-vault-cloud-run-${var.env}"].name
+    name = var.create_vpc ? module.vpc[0].subnets["${local.pvault_region}/${var.deployment_id}-sb-connector-vault-cloud-run-${var.env}"].name : (var.vault_cn_subnet_id ? var.vault_cn_subnet_id : null)
```
Comment (nirhaas) [inline gcp-cloud-run/10-network.tf:124]: Can you move this one and the others that starts with `var.create_vpc` to locals?

```terraform
locals {
	vpc_access_connector_subnet_name = var.create_vpc ? module.vpc[0].subnets["${local.pvault_region}/${var.deployment_id}-sb-connector-vault-cloud-run-${var.env}"].name : (var.vault_cn_subnet_id ? var.vault_cn_subnet_id : null)
}
```

File: gcp-cloud-run/40-cloud-sql.tf [hunk 4]
```diff
@@ -42,7 +43,7 @@ module "postgresql-db" {
 
   ip_configuration = {
     ipv4_enabled        = false
-    private_network     = module.vpc.network_id
+    private_network     = var.create_vpc ? module.vpc[0].network_id : var.vpc_id
```
Comment (nirhaas) [inline gcp-cloud-run/40-cloud-sql.tf:46]: Move to local

File: gcp-cloud-run/50-cloud-run.tf [hunk 5]
```diff
@@ -24,7 +26,14 @@ resource "google_cloud_run_service" "pvault-server" {
       timeout_seconds       = 28
       container_concurrency = 100
       containers {
-        image = "${var.image}:${var.vault_version}"
+        image = "${var.pvault_repository}:${var.pvault_tag}"
+        dynamic "env" {
```
Comment (nirhaas) [inline gcp-cloud-run/50-cloud-run.tf:30]: That's awesome. What would happen though if someone passes `PVAULT_DEVMODE` here and then the cloudrun container have two blocks of
```hcl
env {
	name = "PVAULT_DEVMODE"
	value = "1'
}
...
env {
	name = "PVAULT_DEVMODE"
	value = "1'
}
```

File: gcp-cloud-run/README.md
Comment (nirhaas) [inline gcp-cloud-run/README.md:1]: @david-piiano please review

Comment (david-piiano) [inline gcp-cloud-run/README.md:1]: Thanks @yossi-lavi - the readme changes are minimal and we can move on with that. No problem. FYI: @nirhaas

File: gcp-cloud-run/50-cloud-run.tf (hunk 5 above)
Comment (yossi-lavi) [reply gcp-cloud-run/50-cloud-run.tf:30]: I added in the description the names of all the env vars that won't be applied if put on the map. I put the dynamic before all the envs so the ones we have vars for them will be applied.

**Review by nirhaas — APPROVED** [review]

**Pull request merged**